Libraries/src/Amazon.Lambda.RuntimeSupport/Bootstrap/UserCodeInit.cs

```
@@ -0,0 +1,194 @@
+            }
+        }
+
+        public static void PreJitAssembly(Assembly a)
+        {
+            // Storage to ensure not loading the same assembly twice and optimize calls to GetAssemblies()
+            var loaded = new HashSet<string>();
+
+            PrepareAssembly(a);
+            LoadReferencedAssemblies(a);
+            // Filter to avoid loading all the .net framework
+            bool ShouldLoad(string assemblyName)
+            {
+                return !loaded.Contains(assemblyName) && NotNetFramework(assemblyName);
+            }
+
+            bool NotNetFramework(string assemblyName)
+            {
+                return !assemblyName.StartsWith("Microsoft.")
+                       && !assemblyName.StartsWith("System.")
+                       && !assemblyName.StartsWith("Newtonsoft.")
+                       && !assemblyName.StartsWith("netstandard");
+            }
+
+            void PrepareAssembly(Assembly assembly)
+            {
+                foreach (var type in assembly.GetTypes())
+                {
+                    foreach (var method in type.GetMethods(BindingFlags.DeclaredOnly |
+                                                           BindingFlags.NonPublic |
+                                                           BindingFlags.Public | BindingFlags.Instance |
+                                                           BindingFlags.Static))
+                    {
+                        try
+                        {
+                            if (method.IsAbstract)
+                                continue;
+
+                            RuntimeHelpers.PrepareMethod(
+                                method.MethodHandle);
+
+                        }
+                        catch (Exception e)
+                        {
+                            InternalLogger.GetDefaultLogger().LogError(e, "PreJit: Error with PrepareAssembly");
+                        }
```

How likely is this to happen, and how would one go about addressing it? (this exception is swallowed in .NET Core 3.1 RIC)

---

Probably very unlikely but like since the logging only shows up when debugging Amazon.Lambda.RuntimeSupport. The more logging when debugging this library the better is my opinion.